### DISCUSSION TITLE
Creating MS SQL localhost DB accessibility

### DISCUSSION THREAD

**Discussion by RyuZinOh** [community/community#136768] [Programming Help] [answered]
### Select Topic Area

Question

### Body

how do i like add all my database created on ms -sql  to localhost so that same network shared devices can act as a client host system to manipulate the database, can i use xampp, what the procedure, and is their any built in tools or extension for this?

**Comment by Yackens** [comment]
To share your MS SQL Server databases on a local network so that other devices can connect and manipulate the database, you don’t typically use XAMPP, as XAMPP is more focused on running Apache, MySQL (MariaDB), PHP, and Perl. Instead, you will use Microsoft SQL Server's own features to accomplish this. My advice would be to following these steps:

1. Install SQL Server on the Host Machine
2. Configure SQL Server to Allow Network Connections (In SQL Server Configuration Manager, select SQL Server Network Configuration and I would recommend you to add the TCP protocol, then configure a firewall and allow connections to your server through SQL Server Management Studio - for additional info follow any guide online).
3. Connect to the Database from Other Devices

**Reply by RyuZinOh** [reply]
I  have configured the TCP protocol and set both outbound and inbound rules. However, when attempting to connect to the host server using SQL Server Management Studio (SSMS) from the client, the connection fails. I am wondering if this issue could be due to improper setup of the virtual machine. Unfortunately, I do not have another device on the same network to test the connection so i am using vm. Would the connection likely work on a different device within the same network, assuming the VM setup is not the issue ?

**Reply by Yackens** [reply, marked as answer]
If you've correctly configured the TCP protocol and set up both outbound and inbound firewall rules, the connection should work from any device on the same network, including virtual machines (VMs). However, VMs can introduce some complexity, particularly if they are not correctly configured to communicate with the host machine or the rest of the network.  Check the following things:
- Network Configuration: Ensure that the VM is set to use a network mode that allows it to be on the same network as the host machine
- Testing Connectivity: From the VM use **ping** to test basic connectivity from the VM to the host machine. From the Host Machine you can also try using **telnet** on the VM to test if the SQL Server port (usually 1433) is open and reachable from the VM.
- Windows Firewall: Double-check that Windows Firewall is configured to allow traffic through the SQL Server port. In 80% of the cases, this is the issue.

If you’ve gone through these steps and still face issues, it might be helpful to test with a different device on the network to isolate whether the problem is with the VM setup or the SQL Server configuration. As you mentioned, try the connection with a different device in the same network to see if it works. Hope this helps!

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @RyuZinOh ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.
